### ISSUE TITLE
epoch-api 404

### ISSUE THREAD

**Issue by mokipedia** [postmatic/epoch#39]
I wanted to give epoch a shot, because we do not like to switch to disqus but need a better solution for our comments.
When I installl the plugin on my test server, it starts, the comments are gone and i cannot post a comment anymore. Looking into the error console, it gets quite apparent why:
POST ...my/url/wordpress/epoch-api 404 (Not Found). I found someone else with the same problem, but as I am on 1.0.0 I had assumed for him the problem was resolved:
https://wordpress.org/support/topic/doesnt-work-964

Error Console log:

0
Failed to load resource: the server responded with a status of 404 (Not Found)
Failed to load resource: the server responded with a status of 404 (Not Found)
POST http://[my/server/path]/wordpress/epoch-api 404 (Not Found)
XHR finished loading: POST "http://[my/server/path]/wordpress/epoch-api".
POST http://[my/server/path]/wordpress/epoch-api 404 (Not Found)
XHR finished loading: POST "http://[my/server/path]/wordpress/epoch-api".
POST http://[my/server/path]/wordpress/epoch-api 404 (Not Found)
...
and so on

any help much appreciated!

**Comment by Shelob9** [comment]
Can you try resaving your permalink settings please?

**Comment by mokipedia** [comment]
oh, you were absolutely right! thanks a lot, that saved me quite a bit of time!

**Comment by Shelob9** [comment]
@mokipedia Thanks for reporting back. That shouldn't be necessary,  now I have to hunt down why it was for you.

Which theme are you using?

**Comment by mokipedia** [comment]
@Shelob9 we are using a self written theme.
But i think what happend was a bit of a misconfiguration on my side, i had prettylinks off, only showing the standard permalink /my/page/?post=xyz
maybe that was why it did not work? i hadn't bothered changing as this is only my testserver.

**Comment by tormondo** [comment]
Hi guys,

I have not found any other topics (on the Net!) about problems with missing "epoch-api" folder and decided to late-join your discussion here.

WP 4.4, Twenty Twelve, Epoch with Ratings and Crowd moderation. All latest.

When I am logged out and go to Epoch comments, it offers a link under existing comments "Log in to Reply". When I press it, it takes me to URL "http://DOMAIN_NAME/wp-login.php?redirect_to=http://DOMAIN_NAME/epoch-api" and gives a login form. I login successfully but it then parks me on the 404 page. Of course, I do not have this folder in my HTTP-server structure.

How can I fight this problem down?

And a separate question: can I suppress displaying this invitation when logged-out?

Thank you for your work.
t

**Comment by Shelob9** [comment]
This is an interesting bug. Please open a new issue, as it is not related.

BTW You can solve this using the login_redirect filter
https://codex.wordpress.org/Plugin_API/Filter_Reference/login_redirect

## 

JoshPress.net http://joshpress.net/
CalderaWP.com http://calderawp.com/
IngotHQ.com
@Josh412 http://twitter.com/Josh412

**Comment by tormondo** [comment]
Thanks.

This did not work for me. For now I simply hacked epoch/classes/front/vars.php
static public $endpoint = 'epoch-api'; // have set this var to an empty string
This results in logging in and landing on the main page. At least this.

I did not quickly find how to remove or annul link "Log in to Reply". Presumably, the code takes standard global strings from WP (or somewhere outside Epoch folders).

PS
sorry for not opening a proper thread on it, I'm not very much into development. But if you or anyone feels it's worth doing, please, feel free to replicate the issue in a proper bug-tracker. Sorry for instantaneity once again.

**Comment by tormondo** [comment]
Ohhh... my goodness!

Finally, I found the problem. I had a login redirecting plugin and fixes to help them operate in all relevant URLs of WP. They somehow interfered with Epoch before I turned the code back to virgin pink. Removed them. Now it works with other plugings and without any functions.php fixes to help them out. The correctly working redirection plugins are: **Logout Redirect**, **Sky Login Redirect**, **Login Logout Menu**. For some reasong **Login Redirect**, a counterpart of **Logout Redirect**, did not work in combination with Epoch and made me dive into scripting.

Btw, there are two different ways redirects work: one via an URL argument "...&redirect_to=...", the other without it (Login redirect works this way). As far as I can see, Epoch welcomes the latter and crashes its head against the wall with the former. In principle, the Epoch code can be designed to anticipate either way to be more stable with so differently working plugins (login works via the hidden method, logout via the URL-argument).

Sorry for confusing you all.
Good luck!
t
